source/common/access_log/access_log_formatter.h

```diff
@@ -21,8 +21,12 @@ class AccessLogFormatParser {
   static std::vector<FormatterPtr> parse(const std::string& format);
 
 private:
-  static void parseCommand(const std::string& token, const size_t start, std::string& main_header,
-                           std::string& alternative_header, absl::optional<size_t>& max_length);
+  static void parseCommandHeader(const std::string& token, const size_t start,
+                                 std::string& main_header, std::string& alternative_header,
+                                 absl::optional<size_t>& max_length);
+  static void parseCommand(const std::string& token, const size_t start,
+                           const std::string& separator, std::string& main,
+                           std::vector<std::string>& subs, absl::optional<size_t>& max_length);
```

Comment:
I think some comments on what subs are and the expected pre/postconditions would be helpful.